python/ray/data/impl/batcher.py

```diff
@@ -63,7 +63,7 @@ def next_batch(self) -> Block:
                 leftover.append(block)
             elif accessor.num_rows() <= needed:
                 # We need this entire block to fill out a batch.
-                output.add_block(block)
+                output.add_block(accessor.to_arrow())
```

Comment:
Why do we need to convert block to arrow? Batcher is agnostic to bock type, right?

Reply:
Thanks for the reply, I‘m trying to fix the error reported by `test_raydp_to_torch_iter()`,  `to_arrow()` is really not a good solution, I made new modification after rethinking. As mentioned in the issue above, an `AssertionError` is raised before this change, with the latest commit, this can be fixed.